src/modules/compliance/src/lib/procedures/ensureFilePermissions.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include <Evaluator.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <string.h>
+#include <unistd.h>
+#include <grp.h>
+#include <pwd.h>
+
+
+namespace compliance {
+
+AUDIT_FN(ensureFilePermissions)
+{
+    struct stat statbuf;
+    if (args.find("filename") == args.end())
+    {
+        return Error("No filename provided");
+    }
+    logstream << "ensureFilePermissions for " << args["filename"];
+
+    if (stat(args["filename"].c_str(), &statbuf) < 0)
+    {
+        return Error("Stat error");
+    }
+
+    if (args.find("user") != args.end())
+    {
+        struct passwd *pwd = getpwuid(statbuf.st_uid);
+        if (nullptr == pwd)
+        {
+            logstream << "No user with uid " << statbuf.st_uid;
```

Comment:
why are we not using the OSCONFIG_LOG_INFO/OSCONFIG_LOG_ERROR macros here?

Reply:
The logstream is used to create a string with the result of the audit for the UI. It is then also logged to OSCONFIG_LOG.

Reply:
When you say the UI, do you mean this is whats displayed in the non-compliance reason in the portal? In the end however, wouldn't this string be a very long concatenated string, im not seeing any newlines?

Reply:
Unfortunately - yes, that's the only thing that we can (at least for now) present to the user.
It usually won't be that complex though.